graylog2-server/src/main/java/org/graylog2/rest/resources/system/SessionsResource.java

```diff
@@ -124,11 +126,15 @@ public JsonNode newSession(@Context ContainerRequestContext requestContext,
         final String sessionId = shiroSecurityContext.getUsername();
         final String host = RestTools.getRemoteAddrFromRequest(grizzlyRequest, trustedSubnets);
 
-        Optional<Session> session = sessionCreator.create(sessionId, host, authToken);
-        if (session.isPresent()) {
-            return sessionResponseFactory.forSession(session.get());
-        } else {
-            throw new NotAuthorizedException("Invalid credentials.", "Basic realm=\"Graylog Server session\"");
+        try {
+            Optional<Session> session = sessionCreator.create(sessionId, host, authToken);
+            if (session.isPresent()) {
+                return sessionResponseFactory.forSession(session.get());
+            } else {
+                throw new NotAuthorizedException("Invalid credentials.", "Basic realm=\"Graylog Server session\"");
+            }
+        } catch (AuthenticationServiceUnavailableException e) {
+            throw new ServiceUnavailableException(e.getMessage());
```

Comment:
I think we shouldn't expose too many details to unauthenticated users.
They shouldn't be able to discover which auth backends we are using.
How about
```suggestion
            throw new ServiceUnavailableException("Authentication service unavailable");
```